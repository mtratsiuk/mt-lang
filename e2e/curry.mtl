(def add
  (|a b|
    (+ a b)))

(def array [2 4 8])

(print
  (array/map (add 2)))
