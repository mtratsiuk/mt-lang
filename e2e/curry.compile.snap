const __mtl_curry = f => (...args) => {
  if (args.length >= f.length) {
    return f(...args);
  }

  return __mtl_curry(f.bind(null, ...args));
};

const add = __mtl_curry((a, b) => {
  return (a + b);
});

const array = [2, 4, 8];

console.log(String(array.map(add(2))));
